adapters/repos/db/index.go

```diff
@@ -1724,71 +1728,119 @@ func (i *Index) IncomingDeleteObject(ctx context.Context, shardName string,
 	return shard.DeleteObject(ctx, id)
 }
 
-// func (i *Index) localShard(name string) ShardLike {
-// 	return i.shards.Load(name)
-// }
+func (i *Index) getClass() *models.Class {
+	className := i.Config.ClassName.String()
+	return i.getSchema.ReadOnlyClass(className)
+}
 
-func (i *Index) getLocalShardNoShutdown(shardName string) (ShardLike, func(), error) {
-	i.shardInUseLocks.RLock(shardName)
-	defer i.shardInUseLocks.RUnlock(shardName)
+// Intended to run on "receiver" nodes, where local shard
```

Comment:
Is the godoc comment still relevant to how the function has changed ?

Reply:
the behaviour should be equivalent to the original one, just avoiding some races